compiler/rustc_session/src/config.rs

```diff
@@ -1942,9 +1942,12 @@ fn parse_native_lib_kind(
         "static" => NativeLibKind::Static { bundle: None, whole_archive: None },
         "dylib" => NativeLibKind::Dylib { as_needed: None },
         "framework" => NativeLibKind::Framework { as_needed: None },
+        "link-arg" => NativeLibKind::LinkArg,
```

Comment:
Could you add something like `report_unstable_modifier` below to this place, so that `-l link-arg` is not available in stable compiler releases.